addons/sourcemod/scripting/shavit-tas.sp

```diff
@@ -0,0 +1,1407 @@
+							{
+								if(gI_Type[client] == Type_Override)
+								{
+									return Plugin_Continue;
+								}
+								else if(gI_Type[client] == Type_SurfOverride && IsSurfing(client))
+								{
+									return Plugin_Continue;
+								}
+							}
+
+							float fFowardMove, fSideMove;
+							float fMaxSpeed = GetEntPropFloat(client, Prop_Data, "m_flMaxspeed");
+							float fSurfaceFriction = 1.0;
+							if(gI_SurfaceFrictionOffset > 0)
+							{
+								fSurfaceFriction = GetEntDataFloat(client, gI_SurfaceFrictionOffset);
+								if(gCV_AutoFind_Offset.BoolValue && !(fSurfaceFriction == 0.25 || fSurfaceFriction == 1.0))
+								{
+									FindNewFrictionOffset(client);
+								}
+							}
+
+
+							float fVelocity[3], fVelocity2D[2];
+
+							GetEntPropVector(client, Prop_Data, "m_vecVelocity", fVelocity);
+
+							fVelocity2D[0] = fVelocity[0];
+							fVelocity2D[1] = fVelocity[1];
+
+							// PrintToChat(client, "%f", SquareRoot(fVelocity2D[0] * fVelocity2D[0] + fVelocity2D[1] * fVelocity2D[1]));
+
+							GetIdealMovementsInAir(angles[1], fVelocity2D, fMaxSpeed, fSurfaceFriction, fFowardMove, fSideMove);
+
+							float fAngleDifference = AngleNormalize(angles[1] - gF_LastAngle[client]);
+							float fCurrentAngles = FloatAbs(fAngleDifference);
+
+
+							// Right
+							if (fAngleDifference < 0.0)
+							{
+								float fMaxDelta = GetMaxDeltaInAir(fVelocity2D, fMaxSpeed, fSurfaceFriction, true);
+								vel[1] = gF_MaxMove;
+
+								if (fCurrentAngles <= fMaxDelta * gF_Power[client])
+								{
+									vel[0] = fFowardMove * gF_MaxMove;
+									vel[1] = fSideMove * gF_MaxMove;
+								}
+							}
+							else if (fAngleDifference > 0.0)
+							{
+								float fMaxDelta = GetMaxDeltaInAir(fVelocity2D, fMaxSpeed, fSurfaceFriction, false);
+								vel[1] = -gF_MaxMove;
+
+								if (fCurrentAngles <= fMaxDelta * gF_Power[client])
+								{
+									vel[0] = fFowardMove * gF_MaxMove;
+									vel[1] = fSideMove * gF_MaxMove;
+								}
+							}
+							else
+							{
+								vel[0] = fFowardMove * gF_MaxMove;
+								vel[1] = fSideMove * gF_MaxMove;
+							}
+						}
+					}
+				}
+				/*
+							WIGGLEHACK END
+														*/
+
+				if(gF_TimescaleTicksPassed[client] >= 1.0)
+				{
+					gF_TimescaleTicksPassed[client] -= 1.0;
+
+					gF_TASTime[client] += GetTickInterval();
+
+					int iFrameNumber = gA_Frames[client].Length+1;
+					if(gI_IndexCounter[client] != iFrameNumber-2)
+					{
+						//UnPaused in different tick
+						iFrameNumber = gI_IndexCounter[client]+1;
+					}
+					gA_Frames[client].Resize(iFrameNumber);
+
+					framedata_t frame;
+
+					float fAngles[3];
+					GetClientEyeAngles(client, fAngles);
+					frame.fEyeAngles[0] = fAngles[0];
+					frame.fEyeAngles[1] = fAngles[1];
+
+					GetEntPropVector(client, Prop_Send, "m_vecOrigin", frame.fPosition);
+					GetEntPropVector(client, Prop_Data, "m_vecAbsVelocity", frame.fVelocity);
```

Comment:
Not likely to be an issue, but processmovement uses `m_vecVelocity`, so unless you have a certain reason for using abs it should be updated to m_vecVelocity